templates/server/post-receive.erb

```diff
@@ -111,12 +111,33 @@ $stdin.each_line do |line|
         end
         do_cmd("git submodule sync")
         do_cmd("git submodule update --init --recursive")
+        <% if @git_repo_r10k -%>
+        if File.exists? 'Puppetfile'
+          puts("Installing modules from r10k")
+          do_cmd("r10k puppetfile install")
+        end
+        <% end -%>
+        <% if @git_repo_gen_types -%>
+        puts("Generating types for #{environment_name})"
+        do_cmd("puppet generate types --environment #{environment_name}"
+        <% end -%>
       end
     else
       # Instantiate a new environment from the current repository.
 
       puts "Creating new environment #{environment_name}"
       do_cmd("git clone --recursive #{SOURCE_REPOSITORY} #{environment_path} --branch #{branchname}")
+      <% if @git_repo_r10k -%>
+      Dir.chdir environment_path
+      if File.exists? 'Puppetfile'
+        puts("Installing modules from r10k")
+        do_cmd("r10k puppetfile install")
+      end
+      <% end -%>
+      <% if @git_repo_gen_types -%>
+      puts("Generating types for #{environment_name})"
+      do_cmd("puppet generate types --environment #{environment_name}"
```

Comment:
I recall in the past that I also used an explicit directory path. In theory it shouldn't make a difference, but it doesn't hurt. We already have that path anyway so I wonder if it would be the right thing to do.

Reply:
It seems that `puppet` only excepts environment names (and not full paths).

```
Could not find a directory environment named '/etc/puppetlabs/code/environments/test'
```

Reply:
Filed ticket upstream as this behavior seems odd to me : https://tickets.puppetlabs.com/browse/PUP-11700